routing/payment_lifecycle.go

```diff
@@ -346,6 +356,33 @@ func (p *paymentLifecycle) checkTimeout() error {
 	return nil
 }
 
+// checkContext checks whether the payment context has been cancelled.
+func (p *paymentLifecycle) checkContext(ctx context.Context) error {
+	if ctx == nil {
+		return nil
+	}
```

Comment:
I think it is a bit of an anti-pattern to ever allow passing in a nil context. I think we should just create one at the call-sites where one is not available